analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs

```diff
@@ -29,50 +29,38 @@ public class RuleTypeTest
 
         [TestMethod]
         public void DetectRuleTypeChanges_CS() =>
-            DetectTypeChanges(csharp::SonarAnalyzer.RspecStrings.ResourceManager,
-                              CsRuleTypeMapping.RuleTypesCs,
-                              nameof(CsRuleTypeMapping.RuleTypesCs),
-                              deletedRules);
+            DetectTypeChanges(csharp::SonarAnalyzer.RspecStrings.ResourceManager, RuleTypeMappingCS.Rules, nameof(RuleTypeMappingCS));
 
         [TestMethod]
         public void DetectRuleTypeChanges_VB() =>
-            DetectTypeChanges(vbnet::SonarAnalyzer.RspecStrings.ResourceManager,
-                              VbRuleTypeMapping.RuleTypesVb,
-                              nameof(VbRuleTypeMapping.RuleTypesVb),
-                              deletedRules);
+            DetectTypeChanges(vbnet::SonarAnalyzer.RspecStrings.ResourceManager, RuleTypeMappingVB.Rules, nameof(RuleTypeMappingVB));
 
         [AssertionMethod]
-        private static void DetectTypeChanges(ResourceManager resourceManager,
-                                              IImmutableDictionary<string, string> expectedTypes,
-                                              string expectedTypesName,
-                                              ICollection<string> deletedRules)
+        private static void DetectTypeChanges(ResourceManager resourceManager, IImmutableDictionary<string, string> expectedTypes, string expectedTypesName)
         {
             var items = Enumerable.Range(1, 10000)
-                                  .Select(i => new
+                                  .Select(x => "S" + x)
+                                  .Select(x => new
                                   {
-                                      ExpectedType = expectedTypes.GetValueOrDefault(i.ToString()),
-                                      ActualType = resourceManager.GetString($"S{i}_Type"),
-                                      RuleId = i
+                                      ExpectedType = expectedTypes.GetValueOrDefault(x),
+                                      ActualType = resourceManager.GetString($"{x}_Type"),
+                                      RuleId = x
                                   })
                                   .Where(x => x.ActualType != x.ExpectedType)
                                   .ToList();
 
-            // IMPORTANT: If this test fails, you should add the types of the new rules
-            // in the dictionaries above. It is a manual task, sorry.
-            var newRules = items.Where(x => x.ExpectedType == null);
+            // If this test fails, you should add the types of the new rules in the dictionaries above. It is a manual task, sorry.
+            var newRules = items.Where(x => x.ExpectedType is null);
             newRules.Should().BeEmpty($"you need to add the rules in {expectedTypesName}");
```

Comment:
I can't remember: Does FluentAssertions outputs the entries in the list if `BeEmpty` fails. If not, the message should name the RuleIds that need to be added.

Reply:
Yes, it does.